### PR TITLE
Enabling QueryStore options to the database properties

### PR DESCRIPTION
**Pull request by ssreerama**
This PR enables the query store options tab to the database properties.

QueryStore
QueryStore_disk

### CODE REVIEW

**Review by barbaravaldez — APPROVED** [review]
add validation for max values in the number input box elements

**Review by corivera — APPROVED** [review]

**Comment by erinstellato-ms** [comment]
Any reason that wait statistics capture mode is check box, versus a drop down with On/Off?

**Comment by ssreerama** [comment]
> Any reason that wait statistics capture mode is check box, versus a drop down with On/Off? Also, are we displaying the disk usage anywhere? I know that having it as a pie chart is beyond scope, which is fine, but I was thinking we would display the numbers at the bottom of the pane.

@erinstellato-ms , In ADS properties tabs when we have a true/false DD kind of options we replaced with checkbox. I just followed the same pattern for this options, as it will be either On/Off. Let me know it must be a dd and I can replace it with a dropdown. Thanks :)

**Comment by ssreerama** [comment]
@erinstellato-ms , Need help in understanding the calculations here:

image

I get two values directly from the smo, but need the used value calculations formulae here. 
sai_Test : size
Query store used: ???

Query Store Available: QueryStoreRetention > Max Size (MB)
Query store used: ???

**Comment by erinstellato-ms** [comment]
@ssreerama ok, for current disk usage, the number for sai_Test is the total size of the database.  Does that come from SMO?  For Query Store Used (beneath DB size), it's the amount of space that QS uses...so I suspect it would pull current_storage_size_mb from sys.database_query_store_options.

SELECT current_storage_size_mb
FROM sys.database_query_store_options)

For the second set of data, Query Store Available is max_storage_size_mb - current_storage_size_mb and Query Store Used is current_storage_size_mb.


SELECT 
	current_storage_size_mb AS "Query Store Used",
	(max_storage_size_mb - current_storage_size_mb) AS "Query Store Available"
FROM sys.database_query_store_options

Now...If we are just displaying numbers, I feel like we could do better.  Suggestion below, let me know what you think...

Current Disk Usage

DatabaseName: DBSize
Query Store Used: max_storage_size_mb - current_storage_size_mb
Query Store Available: current_storage_size_mb

**Comment by ssreerama** [comment]
Got it, I get both size and queryStorageused form SMO.

correction: 
DatabaseName: DBSize
Query Store Used: current_storage_size_mb
Query Store Available: max_storage_size_mb - current_storage_size_mb

File: extensions/mssql/src/objectManagement/ui/databaseDialog.ts [hunk 1]
```diff
@@ -1581,6 +1623,245 @@ export class DatabaseDialog extends ObjectManagementDialogBase<Database, Databas
...
+		this.purgeQueryDataButton.width = DefaultInputWidth;
+		await this.purgeQueryDataButton.updateCssStyles({ 'margin': '10px 0px, 0px, 0px' });
+		containers.push(this.createLabelInputContainer('', this.purgeQueryDataButton));
+
+		const diskUsageSection = this.createGroup(localizedConstants.QueryStoreCurrentDiskUsageSectionText, containers, true);
+		this.queryStoreTabSectionsContainer.push(diskUsageSection);
+	}
+
+	/**
+	 *  Opens confirmation warning Clears the query store data for the database
```
Comment (corivera) [inline extensions/mssql/src/objectManagement/ui/databaseDialog.ts:1830]:
```suggestion
	 *  Opens confirmation warning for clearing the query store data for the database
```

**Review by corivera — APPROVED** [review]

**Comment by ssreerama** [comment]
Updated STS version to 4.9.0.34
image

**Pull request merged**